src/input.rs

```diff
@@ -435,54 +435,63 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     }
 
     /// Underline URLs and change the mouse cursor when URL hover state changes.
-    fn update_url_highlight(&mut self, point: Point) {
+    fn update_url_highlight(&mut self, point: Point, modifiers: ModifiersState) {
         let mouse_mode =
             TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG | TermMode::MOUSE_REPORT_CLICK;
 
-        if let Some(Url { text, origin }) = self.ctx.terminal().url_search(point.into()) {
-            let mouse_cursor = if self.ctx.terminal().mode().intersects(mouse_mode) {
-                MouseCursor::Default
-            } else {
-                MouseCursor::Text
-            };
+        // Only show URLs as launchable when all required modifiers are pressed
+        if self.mouse_config.url.modifiers.relaxed_eq(modifiers)
+            && (!self.ctx.terminal().mode().contains(TermMode::ALT_SCREEN) || modifiers.shift)
```

Comment:
Is this internal that in alt screen mode, shift has to be the modifier?  This seems to force me to use shift in tmux :)

BTW, thanks for the work!  I've been wanting this for a long time.

Reply:
I'm not sure what got over me here. This shouldn't check for `ALT_SCREEN`, but instead check for the mouse mode to be enabled. Thanks for pointing it out, current implementation makes no sense at all.

I'm going to include that into https://github.com/jwilm/alacritty/pull/2195.